cpp/include/games/nim/Game.hpp

```diff
@@ -40,15 +43,11 @@ struct Game {
 
   struct State {
     auto operator<=>(const State& other) const = default;
-
-    size_t hash() const {
-      auto tuple = std::make_tuple(stones_left, current_player);
-      std::hash<decltype(tuple)> hasher;
-      return hasher(tuple);
-    }
+    size_t hash() const;
 
     int stones_left;
     int current_player;
+    bool chance_active;
```

Comment:
In my opinion, clearest here would be to replace `bool chance_active` with `core::action_mode_t current_mode`.

In `Constants.hpp`, add:

```
const core::action_mode_t kPlayerMode = 0;
const core::action_mode_t kChanceMode = 1;
```